libs/designer-ui/src/lib/editor/base/utils/keyvalueitem.ts

```diff
@@ -63,7 +63,7 @@ export const convertValueType = (value: ValueSegment[], type?: string): string |
   const stringSegments = convertSegmentsToString(value).trim();
   // checks for known types
   if (
-    (stringSegments.startsWith('{') && stringSegments.endsWith('}')) ||
+    (stringSegments.startsWith('@{') && stringSegments.indexOf('}') === stringSegments.length - 1) ||
```

Comment:
Can we get some test coverage on this function to make sure this produces the result we're looking for and doesn't regress?

Reply:
Added 3 unit tests for the parent function that serializes the input.